scripts/handle-cnab-package.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+PACKAGING_IMAGE="mcr.microsoft.com/container-package-app:latest"
+
+main() {
+    local repo_root
+    repo_root=$(git rev-parse --show-toplevel)
+
+    local changed
+    changed=$(ct list-changed --config "$repo_root/ct.yaml")
+
+    if [[ -z "$changed" ]]; then
+        exit 0
+    fi
+
+    local num_changed
+    num_changed=$(wc -l <<< "$changed")
+
+    if ((num_changed > 1)); then
+        echo "This PR has changes to multiple charts. Please create individual PRs per chart." >&2
+        exit 1
+    fi
+
+    # Strip charts directory.
+    changed="${changed##*/}"
+
+    if [[ "$CNAB_ACTION" != "VERIFY" && "$CNAB_ACTION" != "RELEASE" ]]; then
+        echo "CNAB_ACTION must be one of: [VERIFY, RELEASE]."
+        exit 1
+    fi
+
+    if [[ "$PR_TITLE" != "[$changed] "* ]]; then
+        echo "PR title must start with '[$changed] '." >&2
+        exit 1
+    fi
+
+    if [[ ! -d "./cnab-config/$changed" ]]; then
+        echo "CNAB bundle is not setup for $changed service, skipping.."
+        exit 0
+    fi
+
+    # Update CNAB manifest.yaml to match Helm Chart version.
+    local chartVersion
+    chartVersion=$(grep "version: " "$PWD/charts/$changed/Chart.yaml" | sed 's|version: ||g')
+    echo "Parsed Helm Chart version: $chartVersion"
+    sed -i "s|<chartVersion>|$chartVersion|g" "$PWD/cnab-config/$changed/manifest.yaml"
+
+    # Parse image details from charts/<service>/values.yaml.
+    local imageName
+    local imageTag
+    local imageRegistry
+    local imageDigest
+    local imageLocation
+    # Strip castai- prefix to gen service name.
+    imageName="${changed##*-}"
+    imageTag=$(grep "appVersion: " "$PWD/charts/$changed/Chart.yaml" | sed 's|appVersion: ||g' | sed 's|"||g')
+    imageRegistry=$(grep "repository: " "$PWD/charts/$changed/values.yaml" | sed 's|repository: ||g')
+    imageRegistry=$(echo "${imageRegistry%/$imageName}" | xargs)
+    imageLocation="$imageRegistry/$imageName:$imageTag"
+    # shellcheck disable=SC2086
+    imageDigest=$(docker pull $imageLocation | grep  "Digest: " | sed 's|''Digest: ||g')
+
+    # Update CNAB values.yaml with image details.
+    sed -i "s|<imageDigest>|$imageDigest|g" "$PWD/cnab-config/$changed/values.yaml"
+    sed -i "s|<imageName>|$imageName|g" "$PWD/cnab-config/$changed/values.yaml"
+    sed -i "s|<imageRegistry>|$imageRegistry|g" "$PWD/cnab-config/$changed/values.yaml"
+
+    # Copy Helm Chart folder to CNAB directory and replace Chart values.yaml with CNAB values.yaml.
+    echo "Copying $changed Helm chart to CNAB directory for packaging"
+    PWD=$(pwd)
+    cp -R "$PWD/charts/$changed" "$PWD/cnab-config/$changed/"
+    mv "$PWD/cnab-config/$changed/values.yaml" "$PWD/cnab-config/$changed/$changed/values.yaml"
```

Comment:
the `$changed/$changed` part is pretty triggering, what is going on there? $changed is a chart name (castai-agent) in this case, why we need "castai-agent/castai-agent" dir

Reply:
ah, it's because of that earlier copy.. would be a bit easier to read if the final structure would not end up as "castai-agent/castai-agent", something like

```
# copy the chart and apply overrides
cp -R "./charts/$changed" "./cnab-config/$changed"
cp -R "./charts/$changed/overrides/values.yaml" "./cnab-config/$changed/values.yaml"
```

Reply:
We need to have Helm chart `castai-agent` and CNAB config files  from `./cnab-config/castai-agent`in one directory for CNAB packaging command to work.

This achieves this:
Directory view
```
❯ tree -d cnab-config
cnab-config
└── castai-agent
    └── castai-agent
        ├── ci
        └── templates

```

File view:
```
❯ tree cnab-config
cnab-config
├── README.md
└── castai-agent
    ├── castai-agent
    │   ├── Chart.yaml
    │   ├── ci
    │   │   ├── secret-ref-values.yaml
    │   │   └── test-values.yaml
    │   ├── templates
    │   │   ├── _helpers.tpl
    │   │   ├── clustervpa-configmap.yaml
    │   │   ├── clustervpa-deployment.yaml
    │   │   ├── deployment.yaml
    │   │   ├── namespace.yaml
    │   │   ├── rbac.yaml
    │   │   ├── resource-quota.yaml
    │   │   ├── secret.yaml
    │   │   └── test-secret.yaml
    │   └── values.yaml
    ├── createUIDefinition.json
    ├── mainTemplate.json
    └── manifest.yaml

```

Reply:
but if we need a specific file structure for things like "cpa verify", then would just be better to create a staging area and copy things over there

```
mkdir .cpa-stage
cp -R "charts/changed" ./cpa-stage/
cp -r "cnab-config/$changed" ./cpa-stage/

.. docker run -v "./cnab-stage"
```

Reply:
Agree, will do that

Reply:
As suggested, created staging area for "cpa verify/build".